metaflow/plugins/logs_cli.py

```diff
@@ -0,0 +1,292 @@
+                        echo_unicode(line.msg)
+                    maybe_old_style = False
+                elif maybe_old_style:
+                    # if they are not available, we may be looking at
+                    # a legacy run (unless we have seen new-style data already
+                    # for another stream). This return an empty string if
+                    # nothing is found
+                    log = ds.load_log_legacy(stream)
+                    if log and timestamps:
+                        raise CommandException(
+                            "We can't show --timestamps for old runs. Sorry!"
+                        )
+                    echo_unicode(log, nl=False)
+    else:
+        raise CommandException(
+            "No Tasks found at the given path -- "
+            "either none exist or none have started yet"
+        )
+
+
+@logs.command(
+    help="Scrub stdout/stderr produced by a task or all tasks in a step. "
+    "The format for input-path is either <run_id>/<step_name> or "
+    "<run_id>/<step_name>/<task_id>."
+)
+@click.argument("input-path")
+@click.option(
+    "--stdout/--no-stdout",
+    default=False,
+    show_default=True,
+    help="Show stdout of the task.",
+)
+@click.option(
+    "--stderr/--no-stderr",
+    default=False,
+    show_default=True,
+    help="Show stderr of the task.",
+)
+@click.option(
+    "--both/--no-both",
+    default=True,
+    show_default=True,
+    help="Show both stdout and stderr of the task.",
+)
+@click.pass_obj
+def scrub(obj, input_path, stdout=None, stderr=None, both=None):
+    types = set()
+    if stdout:
+        types.add("stdout")
+        both = False
+    if stderr:
+        types.add("stderr")
+        both = False
+    if both:
+        types.update(("stdout", "stderr"))
+
+    streams = list(sorted(types, reverse=True))
+
+    # Pathspec can either be run_id/step_name or run_id/step_name/task_id.
+    parts = input_path.split("/")
+    if len(parts) == 2:
+        run_id, step_name = parts
+        task_id = None
+    elif len(parts) == 3:
+        run_id, step_name, task_id = parts
+    else:
+        raise CommandException(
+            "input_path should either be run_id/step_name "
+            "or run_id/step_name/task_id"
+        )
+
+    datastore_set = TaskDataStoreSet(
+        obj.flow_datastore, run_id, steps=[step_name], allow_not_done=True
+    )
+
+    if task_id:
+        ds_list = [
+            TaskDataStore(
+                obj.flow_datastore,
+                run_id=run_id,
+                step_name=step_name,
+                task_id=task_id,
+                mode="d",
+                allow_not_done=True,
+            )
+        ]
+    else:
+        ds_list = list(datastore_set)
+
+    if ds_list:
+        for ds in ds_list:
+            for stream in streams:
+                ds.scrub_logs(LOG_SOURCES, stream)
+        echo("Logs have been scrubbed.")
+
+    else:
+        raise CommandException(
```

Comment:
what is the behavior for tasks that haven't finished yet? a few different cases here - 
1. the attempt is still in flight
2. the attempt has finished but we expect further attempts

Reply:
one approach would be that we instead init the datastore with `allow_not_done=False` ?

Reply:
bunch of fixes regarding this. How things work now:
we rely on `allow_not_done=False` and check when initializing the datastore in `mode="d"` that the latest / specified attempt is done, otherwise we raise an error.

- trying to `logs scrub [pathspec]` on a running task will lead to an error.
- `logs scrub [pathspec] --attempt x` will work even on a running task, as long as the specified attempt has finished.

Reply:
Note: I also tried simply allowing the scrubbing of logs for running tasks, but due to how logs are written in bursts from a buffer, the effects of overwriting a log with the `[REDACTED]` message were being nullified, as every log write cycle would simply rewrite the scrubbed log lines if they were still in buffer.

Reply:
I would definiltely not advocate for allowing the redaction of in-flight logs. THere are just too many complications. We would have to prevent writing to a log that has been redacted which we can't do with past versions of metaflow so there would be this big gapping hole.

Reply:
I like the current behavior described @saikonen -- ie: can scrub past completed attempts with `--attempt`; can scrub latest attempt if finished with no arg and can't scrub any other case.